### PR TITLE
Add openshift.io/build-config.name label to builds

### LINKED ISSUE #4810: Build Labels are not namespaced
After creating a build, if you look at the contents:
```sh
$ oc get build openldap-2 -o json
{
    "kind": "Build",
    "apiVersion": "v1",
    "metadata": {
        "name": "openldap-2",
        "namespace": "ldap",
        "selfLink": "/oapi/v1/namespaces/ldap/builds/openldap-2",
        "uid": "9c4bd9e5-63fa-11e5-94a4-54ee753e2644",
        "resourceVersion": "3674",
        "creationTimestamp": "2015-09-26T02:59:31Z",
        "labels": {
            "buildconfig": "openldap",
            "name": "openldap"
        },
        "annotations": {
            "openshift.io/build.number": "2"
        }
    },
```

Labels are not namespaced with `openshift.io`, whereas the annotations are. Should both be?
cc: @bparees @csrwng

### PR DESCRIPTION
**Pull request by php-coder**
@bparees @mfojtik PTAL

I didn't update `pkg/build/api/v1/types.go` and `pkg/build/api/v1beta3/types.go` files. I'm not sure about them.

Also may be I also should update/fix something in UI because if this change? May be selectors or something else?

Fixes #4810

### CODE REVIEW

File: pkg/build/api/types.go [hunk 1]
```diff
@@ -401,9 +401,14 @@ type BuildOutput struct {
...
-// BuildConfigLabel is the key of a Build label whose value is the ID of a BuildConfig
-// on which the Build is based.
-const BuildConfigLabel = "buildconfig"
+const (
+	// BuildConfigLabel is the key of a Build label whose value is the ID of a BuildConfig
+	// on which the Build is based.
+	BuildConfigLabel = "openshift.io/build-config.name"
+	// OldBuildConfigLabel was used as BuildConfigLabel before adding namespaces.
+	// We keep it for backward compatibility.
+	OldBuildConfigLabel = "buildconfig"
```
Comment (bparees) [inline pkg/build/api/types.go]: call it DeprecatedBuildConfigLabel

**Comment by bparees** [comment]
> I didn't update pkg/build/api/v1/types.go and pkg/build/api/v1beta3/types.go files. I'm not sure about them.

you should actually go ahead and remove those constants, nothing should be referencing them there.

**Comment by php-coder** [comment]
@bparees I've fixed all your comments.

**Comment by mfojtik** [comment]
LGTM [merge]

**Pull request merged**